pkg/collector/corechecks/servicediscovery/usm/service.go

```diff
@@ -91,18 +91,14 @@ func NewDetectionContext(logger *zap.Logger, args []string, envs []string, fs fs
 }
 
 // workingDirFromEnvs returns the current working dir extracted from the PWD env
-func workingDirFromEnvs(envs []string) (string, bool) {
+func workingDirFromEnvs(envs map[string]string) (string, bool) {
 	return extractEnvVar(envs, "PWD")
 }
 
-func extractEnvVar(envs []string, name string) (string, bool) {
-	value := ""
-	prefix := name + "="
-	for _, v := range envs {
-		if strings.HasPrefix(v, prefix) {
-			_, value, _ = strings.Cut(v, "=")
-			break
-		}
+func extractEnvVar(envs map[string]string, name string) (string, bool) {
```

Comment:
should probably remove this function entirely.

Reply:
Leaving this change out of this PR since it's not strictly related to its purpose.